theme/templatetags/sponsor_tags.py

```diff
@@ -4,12 +4,22 @@
 
 register = template.Library()
 
+def preprocess_sponsors(sponsors):
```

Comment:
Converting list of sponsor to dictionary where key is sponsor type and value is sponsors of given type.